src/module/item/physical/document.ts

```diff
@@ -318,6 +319,106 @@ abstract class PhysicalItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | n
         return JSON.stringify(thisData) === JSON.stringify(otherData);
     }
 
+    /** Combine this item with a target item if possible */
+    async stackWith(targetItem: PhysicalItemPF2e): Promise<boolean> {
+        if (this.isStackableWith(targetItem)) {
+            const stackQuantity = this.quantity + targetItem.quantity;
+            if (await this.delete({ render: false })) {
+                await targetItem.update({ "system.quantity": stackQuantity });
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Move this item somewhere else in the inventory, possibly before or after another item or in or out of a container.
+     * If this item and the target item are stackable they will be stacked automatically
+     * @param options Options to control where this item is moved to
+     * @param options.relativeTo An optional target item to sort this item relative to
+     * @param options.sortBefore Should this item be sorted before or after the target item?
+     * @param options.toContainer An optional container to move this item into. If the target item is in a container this can be omitted
+     * @param options.render Render the update? Overridden by moving the item in or out of a container. Defaults to true
+     * @returns
+     */
+    async move({
+        relativeTo,
+        sortBefore,
+        toContainer,
+        render = true,
+    }: {
+        relativeTo?: PhysicalItemPF2e;
+        sortBefore?: boolean;
+        toContainer?: ContainerPF2e<ActorPF2e> | null;
+        render?: boolean;
+    }): Promise<void> {
+        if (!this.actor) {
+            throw ErrorPF2e(`Tried to move an unonwned item!`);
+        }
+        if (!toContainer && !relativeTo) {
+            throw ErrorPF2e(`You must provide either the relativeTo or the toContainer parameter!`);
+        }
+        if (relativeTo) {
+            if (!relativeTo.isOfType("physical")) {
+                throw ErrorPF2e(`The relativeTo item is not a physical item!`);
+            } else if (!relativeTo.actor) {
+                throw ErrorPF2e(`The relativeTo item is not an owned item!`);
+            }
+
+            // If theses two items are stackable, combine them
+            if (await this.stackWith(relativeTo)) return;
+        }
+        if (toContainer) {
+            if (!toContainer.isOfType("backpack")) {
+                throw ErrorPF2e(`The toContainer item is not a container item!`);
+            } else if (!toContainer.actor) {
+                throw ErrorPF2e(`The toContainer item is not an owned item!`);
+            }
+        }
```

Comment:
I suspect most of this could be handled with type checking

Reply:
This is intended for modules or macros that might pass in the wrong item types.